[MIG] stock_request: Migration to 16.0

nicomacr opened the pull request.

JordiBForgeFlow committed:
add stock_request in 11.0

JordiBForgeFlow committed:
Limit the routes available for selection.

etobella committed:
[FIX] stock_request: Add the option to select all the possible rules of a location, the rules of the parents are also allowed

etobella committed:
[IMP] stock_request: Add stock request order

MiquelRForgeFlow committed:
[FIX] Various fixes:
- Error with the sequence number.
- Visible texts that should be in uppercases.
- order_id should only be visible if group_stock_request_order
  option is enabled.
- adds more tests
- adds consistency between models company-wise

aisopuro committed:
[11.0][IMP] stock_request: create request order from products

This commit allows the user to select multiple products from
a product list view and to create a Stock Request Order with
an action. If the user selects templates, all of those templates'
variants will be added to the order.

This is useful as it allows the user to filter products beforehand
based on their stocks, and to create a request order for all
those products whose stock is too low.

etobella committed:
[ADD] stock_request_kanban

JordiBForgeFlow committed:
various fixes

etobella committed:
[IMP] simplify queries

JordiBForgeFlow committed:
[fix] stock_requet: add menu image for stock requests

AaronHForgeFlow committed:
[ADD]stock_request_analytic

LoisRForgeFlow committed:
[11.0][IMP] stock_request: allow to request from all locations types. (#523)

HviorForgeFlow committed:
[11.0][FIX] solve unique constraint problems with product-attribute/product_sequence

HviorForgeFlow committed:
[11.0][FIX] stock_request: added readonly true on related company fields

Commit:
Skipping potential error when move has been deleted

Commit:
Increasing version number

Commit:
Updating contributors

Commit:
Updating version numbering as it was a minor fix

murtuzasaleh committed:
[MIG] stock_request v12

malinkienke committed:
Translated using Weblate (German)

Currently translated at 57.1% (72 of 126 strings)

Translation: stock-logistics-warehouse-12.0/stock-logistics-warehouse-12.0-stock_request
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-12-0/stock-logistics-warehouse-12-0-stock_request/de/

etobella committed:
[IMP] stock_request: Improve cancellation detection

etobella committed:
[IMP] stock_request:
- fixes the computation of the allocated product quantity when a stock move has multiple requests.

etobella committed:
[IMP] stock_request: Allow to set to draft multiple records

max3903 committed:
[FIX] stock_request: Losing partner_id and origin (#664)

HviorForgeFlow committed:
stock_request: Add order desc for SR and SRO

HviorForgeFlow committed:
[12.0][IMP] stock_request:
* Add readme autogenerated
* Add some hooks
* Improve test inheritance (reduce testing time)
* Don't allow qty <= 0
* SR expected_date = order expected_date if SRO

HviorForgeFlow committed:
Allow stock_request_submit to be installed from stock_request

HviorForgeFlow committed:
[FIX] stock_request: Inventory users now can validate Transfers from SRs _compute_qty

MiquelRForgeFlow committed:
[IMP] Simplify inheritance of states

max3903 committed:
[FIX] stock_request: Unable to activate the virtual location

HviorForgeFlow committed:
[FIX] stock_request: add expected_date implicit on tests to avoid errors

rschnapka committed:
Translated using Weblate (German)

Currently translated at 80.9% (131 of 162 strings)

Translation: stock-logistics-warehouse-12.0/stock-logistics-warehouse-12.0-stock_request
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-12-0/stock-logistics-warehouse-12-0-stock_request/de/

HviorForgeFlow committed:
[FIX] stock_request: Remove Inventory User bypass, not needed in v12

LoisRForgeFlow committed:
[12.0][FIX] stock_request: stock request lines were not being
copied when duplicating a stock request order.

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 3.1% (5 of 162 strings)

Translation: stock-logistics-warehouse-12.0/stock-logistics-warehouse-12.0-stock_request
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-12-0/stock-logistics-warehouse-12-0-stock_request/zh_CN/

liweijie0812 committed:
Translated using Weblate (Chinese (Simplified))

Currently translated at 100.0% (162 of 162 strings)

Translation: stock-logistics-warehouse-12.0/stock-logistics-warehouse-12.0-stock_request
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-12-0/stock-logistics-warehouse-12-0-stock_request/zh_CN/

LoisRForgeFlow committed:
[11.0][IMP] stock_request: new icon

Change icon to avoid misunderstanding with purchase_request which
until now had the same icon.

LoisRForgeFlow committed:
[12.0][IMP] stock_request: adapt icon to new style

max3903 committed:
[FIX] stock_request

Commit:
Translated using Weblate (Spanish)

Currently translated at 100.0% (159 of 159 strings)

Translation: stock-logistics-warehouse-12.0/stock-logistics-warehouse-12.0-stock_request
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-12-0/stock-logistics-warehouse-12-0-stock_request/es/

HviorForgeFlow committed:
[IMP] stock_request: black, isort

HviorForgeFlow committed:
[MIG] stock_request: Migration to 13.0

HviorForgeFlow committed:
[13.0][ADD] stock_request_mrp

HviorForgeFlow committed:
[IMP] stock_request:
* Add search filters to Stock Request
* Use SR Manager on action_confirm from testing for better extensibility

Commit:
[IMP] stock_request: Views

LoisRForgeFlow committed:
[13.0][FIX] stock_request: wrong module category reference

murtuzasaleh committed:
[12.0][IMP] Improved _compute_qty code

MiquelRForgeFlow committed:
[UPD] Eficent -> ForgeFlow

celm1990 committed:
FIX: assign value to field compute always
on V13 fields compute must be assign value for avoid cache errors

Commit:
[12.0][FIX] stock_request: fix archived search view

kittiu committed:
Ensure to set done, when qty_cancelled > 0

kittiu committed:
Fix test to work with mrp installed in this repo

kittiu committed:
[IMP] : black, isort, prettier

kittiu committed:
[14.0][MIG] stock_request

kittiu committed:
[14.0][FIX] stock_request, window action access error

JrAdhoc committed:
Translated using Weblate (Spanish)

Currently translated at 92.3% (157 of 170 strings)

Translation: stock-logistics-warehouse-14.0/stock-logistics-warehouse-14.0-stock_request
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-14-0/stock-logistics-warehouse-14-0-stock_request/es/

newtratip committed:
Change parent of model category Stock Request

newtratip committed:
[14.0][FIX] stock_request: Stock Request User can delete Stock Request and Stock Request Order

nilshamerlinck committed:
[FIX] stock_request: default value should not depend on another field

BernatPForgeFlow committed:
[IMP] stock_request: black, isort, prettier

BernatPForgeFlow committed:
[MIG] stock_request: Migration to 15.0

Commit:
[UPD] Update stock_request.pot

OCA-git-bot committed:
[UPD] README.rst

victoralmau committed:
[IMP] stock_request + stock_request_submit: Show error in stock.request.order in the "Confirm" or "Submit" buttons if there are no items.

TT39761

victoralmau committed:
[IMP] stock_request_mrp: Refactor tests

nicomacr committed:
[FIX] stock_request: fix access on act_window

OCA-git-bot committed:
stock_request 15.0.1.0.1

Commit:
[UPD] Update stock_request.pot

OCA-git-bot committed:
stock_request 15.0.1.0.2

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-warehouse-15.0/stock-logistics-warehouse-15.0-stock_request
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-15-0/stock-logistics-warehouse-15-0-stock_request/

nicomacr committed:
[FIX] stock_request: fix access on act_window

OCA-git-bot committed:
stock_request 15.0.1.0.3

Commit:
[UPD] Update stock_request.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-warehouse-15.0/stock-logistics-warehouse-15.0-stock_request
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-15-0/stock-logistics-warehouse-15-0-stock_request/

victoralmau committed:
[IMP] stock_request: Add domain to some fields to prevent deprecate methods with domain return

TT41080

Commit:
[UPD] Update stock_request.pot

OCA-git-bot committed:
stock_request 15.0.1.1.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-warehouse-15.0/stock-logistics-warehouse-15.0-stock_request
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-15-0/stock-logistics-warehouse-15-0-stock_request/

pedrobaeza committed:
[FIX] stock_request: Proper multi-company rules

It includes migration scripts.

victoralmau committed:
[IMP] stock_request: Change the label string to make it consistent

TT41343

Commit:
[UPD] Update stock_request.pot

OCA-git-bot committed:
stock_request 15.0.1.2.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-warehouse-15.0/stock-logistics-warehouse-15.0-stock_request
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-15-0/stock-logistics-warehouse-15-0-stock_request/

victoralmau committed:
[FIX] stock_request: Change state fields to display translated by language

TT41343

Commit:
[UPD] Update stock_request.pot

OCA-git-bot committed:
stock_request 15.0.1.3.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-warehouse-15.0/stock-logistics-warehouse-15.0-stock_request
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-15-0/stock-logistics-warehouse-15-0-stock_request/

pedrobaeza commented:
Please include https://github.com/OCA/stock-logistics-warehouse/pull/1628, although the migration scripts are not needed this time.

nicomacr commented:
> Please include [OCA/stock-logistics-warehouse#1628](https://github.com/OCA/stock-logistics-warehouse/pull/1628), although the migration scripts are not needed this time.

Done! @pedrobaeza

pedrobaeza commented on migrations/15.0.1.4.0/noupdate_changes.xml:
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8" ?>
```
You haven't removed the migration scripts from this commit.

pedrobaeza commented on .pre-commit-config.yaml:
```diff
@@ -101,7 +101,7 @@ repos:
         args: ["--keep-percent-format"]
   - repo: https://github.com/PyCQA/isort
-    rev: 5.10.1
+    rev: 5.12.0
```
Why changing this manually? Let me instead update the copier template

pedrobaeza replied:
#2 Rebase when I merge it.

pedrobaeza replied:
Done

pedrobaeza commented:
I observe a problem in the history fetching that will affect later blaming. I think the problem comes from using the command:
```
git format-patch --keep-subject --stdout origin/16.0..origin/15.0 -- $module | git am -3
```
instead of:
```
git format-patch --keep-subject --stdout origin/16.0..origin/15.0 -- $module | git am -3 --keep --directory $module
```

I think it should be tackled.

nicomacr commented:
> I observe a problem in the history fetching that will affect later blaming. I think the problem comes from using the command:
> 
> ```
> git format-patch --keep-subject --stdout origin/16.0..origin/15.0 -- $module | git am -3
> ```
> 
> instead of:
> 
> ```
> git format-patch --keep-subject --stdout origin/16.0..origin/15.0 -- $module | git am -3 --keep --directory $module
> ```
> 
> I think it should be tackled.

Understood, what actions do you suggest to do? do again the migration of the module using that command?

pedrobaeza commented:
You can:

- Save current module folder to any place.
- Start from scratch doing the procedure.
- Remove the module folder and copy the stored one.
- Commit such changes.
- Forced push to this branch.

nicomacr commented:
@pedrobaeza Ready the changes

pedrobaeza commented:
Ouch, now you face the contrary one: some contents were put inside a double folder. I don't understand why it wasn't correct then nor now...

rousseldenis commented on stock_request/models/stock_request.py:
```diff
@@ -21,12 +34,7 @@ def _get_expected_date():
-            ("done", "Done"),
-            ("cancel", "Cancelled"),
-        ],
+        selection=_get_request_states,
```
Prefer a lambda here to help override and put fileds definition on top

pedrobaeza replied:
Please don't. No need of that to insert elements where you want. See https://github.com/OCA/stock-logistics-warehouse/blob/4e338d152d79d99e298ccf34f548515a3a7aa8f7/stock_request_submit/models/stock_request.py#L11

and if you use a method, the translations are very bad handled.

rousseldenis commented on stock_request/models/stock_request.py:
```diff
@@ -5,13 +5,26 @@
 from odoo.tools import float_compare
 
+REQUEST_STATES = [
+    ("draft", "Draft"),
```
If you don't use selection attribute directly, you should use the translate function.

But I see this approach has been finally removed : https://github.com/OCA/stock-logistics-warehouse/pull/1622/files

So, please get the previous code back

pedrobaeza replied:
The same stated above.

rousseldenis replied:
@nicomacr Please changes this according to what pointed above:

![image](https://user-images.githubusercontent.com/19529533/232504096-82c19201-a6d5-405d-871b-e056aade735f.png)

rousseldenis commented on stock_request/models/stock_request_abstract.py:
```diff
@@ -238,23 +231,33 @@ def onchange_location_id(self):
+    @api.onchange("allow_virtual_location")
+    def onchange_allow_virtual_location(self):
+        if self.allow_virtual_location:
+            return {"domain": {"location_id": []}}
```
I'm not in favor reintroducing domains in onchanges as this is deprecated in core and will be removed in later version.

pedrobaeza replied:
I agree here. This should be handed on a static domain, but you can use simple expressions on domains. Example: https://github.com/OCA/rma/blob/e0c17a60e152f8814242e026166e1f2114789944/rma_sale/models/rma.py#L41-L42

rousseldenis commented on stock_request/models/stock_request_abstract.py:
```diff
@@ -238,23 +231,33 @@ def onchange_location_id(self):
             )
             self.onchange_warehouse_id()
 
+        return {"domain": {"warehouse_id": [("company_id", "=", self.company_id.id)]}}
```
Same as above

rousseldenis reviewed (changes requested):
I would say domains in onchanges is the major thing to not reintroduce

pedrobaeza committed:
[FIX] stock_request: Proper path for noupdate in migration script

victoralmau committed:
[IMP] stock_request: Set with cancelled state the stock moves that are cancelled.

Example use case:

- Create a stock.request (or stock.request.order)
- Confirm.
- Cancel the stock move.
- stock.request and stock.request.order must be set with status cancelled.

TT41827

[IMP] stock_request_mrp: Add tests to check cancel stock request order if production is cancelled.

TT41827

macagua committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (177 of 177 strings)

Translation: stock-logistics-warehouse-15.0/stock-logistics-warehouse-15.0-stock_request
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-warehouse-15-0/stock-logistics-warehouse-15-0-stock_request/es/

kanda999 committed:
Added translation using Weblate (Japanese (ja_JP))

yostashiro committed:
Added translation using Weblate (Japanese)

rousseldenis commented on stock_request/models/stock_request.py:
```diff
@@ -0,0 +1,375 @@
+
+    @api.constrains("order_id", "picking_policy")
+    def check_order_picking_policy(self):
+        if self.order_id and self.order_id.picking_policy != self.picking_policy:
```
@nicomacr For every constraint, change to multi recordset

OCA-git-bot committed:
stock_request 15.0.1.5.0

victoralmau committed:
[FIX] stock_request: Do not change stock.group_stock_user to avoid inconsistency

TT43317

OCA-git-bot committed:
stock_request 15.0.1.6.0

victoralmau committed:
[IMP] stock_request: Apply sudo to prevent requests ACL errors if the user does not have permissions (example: productions).

TT43655

OCA-git-bot committed:
stock_request 15.0.1.6.1

OriolMForgeFlow committed:
[FIX] stock_request: set stock request order state to done when partially delivered

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
stock_request 15.0.1.6.2

rousseldenis commented:
@nicomacr I still observe a problem with the commit history about the directory structure.

Could you please redo migration procedure from scratch ? Thanks

rousseldenis commented:
@nicomacr

victoralmau committed:
[IMP] stock_request: Convert the state field of stock.request.order as compute store to complete all use cases.

This change prevents stock.request.order from remaining in the open state when they have already finished.

TT44811

OCA-git-bot committed:
[UPD] README.rst

EvaSForgeFlow commented:
Hi @nicomacr, I think when you access to Settings you are not able to see the name of the module:

![error_on_config](https://github.com/OCA/stock-logistics-request/assets/131855559/6d06732d-c422-43ee-bb84-c709417b981f)

To avoid this, in the **res_config_settings_views.xml** file include in here:

`<div
                    class="app_settings_block"
                    data-string="Stock Request"
                    data-key="stock_request"
                    groups="stock_request.group_stock_request_manager"
                >`

the module string, as follows:

`<div class="app_settings_block" data-string="Stock Request" string="Stock Request" data-key="stock_request" groups="stock_request.group_stock_request_manager">`

![image](https://github.com/OCA/stock-logistics-request/assets/131855559/5d1ed580-a001-48b9-9436-b4cdd1fa53b7)

nicomacr commented:
@rousseldenis @pedrobaeza sorry for the delay. I've just updated this PR with a new code from scratch.

OCA-git-bot committed:
stock_request 15.0.1.6.3

OCA-git-bot committed:
[UPD] README.rst

JordiMForgeFlow commented:
Hi @nicomacr 

Could you rebase the PR? Thank you! :)

rousseldenis commented:
@nicomacr The correct command to move a module from a repo to another should be:

We assert 'origin' is your branch repository based on stock-logistics-request.

Add the OCA repository in your configuration:

`git remote add OCA https://github.com/OCA/stock-logistics-request`
`git fetch OCA`

Then, here is the trick. Add the former repository remote:

`git remote add OCA-warehouse https://github.com/OCA/stock-logistics-warehouse`
`git fetch OCA-warehouse`

Then do the 'normal' migration command, adapting the source remote:

` git format-patch --keep-subject --stdout OCA/16.0..OCA-warehouse/15.0 -- stock_request | git am -3 --keep`


And voilà! No need to create a temporary directory

rousseldenis commented:
I assert that before doing those commands, you have a clean branch reset on OCA/16.0

nicomacr commented:
@rousseldenis Yes I've made this using that command. I've just fixed the problems now. It's all green

rousseldenis commented:
Mmmmh I don't think so as I see weird commit :

![image](https://github.com/OCA/stock-logistics-request/assets/19529533/6e57eefd-dad1-4c63-8a3e-2dcbddaaa51f)

rousseldenis commented:
And in migration commit, you have several files locations changes :

![image](https://github.com/OCA/stock-logistics-request/assets/19529533/ffd97f09-53ce-4f29-ac48-961105999f7a)

nicomacr committed:
[MIG] stock_request: Migration to 16.0

nicomacr commented:
@rousseldenis Yes, You're right about it, I've had a dirty repo. Now I did it again. and I think it's OK. Please tell me if you see that is OK or not.

matt454357 commented:
Hi @rousseldenis,
I believe the requested changes have been made. Can this PR be merged?

rousseldenis reviewed (approved):
Code review. @nicomacr Next migration, put pre-commit stuff in a separate commit.

rousseldenis commented:
@JordiBForgeFlow @pedrobaeza Maybe you want to review this ?

victoralmau commented on stock_request/models/stock_location_route.py:
```diff
@@ -6,7 +6,7 @@
 
 class StockLocationRoute(models.Model):
-    _inherit = "stock.location.route"
+    _inherit = "stock.route"
```
Rename the class and file name

victoralmau commented on stock_request/tests/test_stock_request.py:
```diff
@@ -1,38 +1,33 @@
 
 from odoo import exceptions, fields
-from odoo.tests import common, new_test_user
+from odoo.tests import common
```
Do not delete `new_test_user` (it is quite useful and you can define as many fields as you want) to create `_create_user()`

victoralmau commented on stock_request/tests/test_stock_request.py:
```diff
@@ -147,23 +182,10 @@ def _create_product_attribute_value(self, name, attribute):
-                **vals
-            )
-        )
-
```
Is it necessary to eliminate these methods?

victoralmau commented on stock_request/tests/test_stock_request.py:
```diff
@@ -0,0 +1,1231 @@
+
+class TestStockRequestBase(TestStockRequest):
+    def setUp(self):
+        super(TestStockRequestBase, self).setUp()
```
```suggestion
        super().setUp()
```

victoralmau commented on stock_request/tests/test_stock_request.py:
```diff
@@ -0,0 +1,1231 @@
+
+
+class TestStockRequest(common.TransactionCase):
+    def setUp(self):
```
```suggestion
    @classmethod
    def setUpClass(cls):
```

victoralmau reviewed (changes requested):
Please remove migration scripts from 15.0

I have checked how it works in runboat, but it is not possible to create a stock request if the user does not have units of measure permissions (the `product_uom_id` field is mandatory but not self-defined, something that in v15 worked correctly). can you check it?

pedrobaeza commented:
> I have checked how it works in runboat, but it is not possible to create a stock request if the user does not have units of measure permissions (the `product_uom_id` field is mandatory but not self-defined, something that in v15 worked correctly). can you check it?

That's because the migration guide hasn't been followed, specifically https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-16.0/_compare/ad52290cdb5f5385ae25fad0c4862a4fa244d9f8...e2d0ba6cd1bfdc8cee787c3fd61b6feecfbe49cc

victoralmau commented on stock_request/views/stock_request_order_views.xml:
```diff
@@ -0,0 +1,214 @@
+                                    <field name="name" readonly="1" />
+                                    <field name="product_id" />
+                                    <field name="allowed_uom_categ_id" invisible="1" />
+                                    <field
+                                        name="product_uom_id"
+                                        options="{'no_open': True, 'no_create': True}"
+                                        groups="uom.group_uom"
+                                    />
```
```suggestion
                                    <field
                                        name="product_uom_id"
                                        options="{'no_open': True, 'no_create': True}"
                                        groups="uom.group_uom"
                                    />
                                    <field name="product_uom_id" readonly="True" groups="!uom.group_uom" force_save="1"/>
```

victoralmau commented on stock_request/views/stock_request_views.xml:
```diff
@@ -0,0 +1,260 @@
+                            <label for="product_uom_qty" />
+                            <div>
+                                <field name="product_uom_qty" class="oe_inline" />
+                                <field
+                                    name="product_uom_id"
+                                    class="oe_inline"
+                                    options="{'no_open': True, 'no_create': True}"
+                                    groups="uom.group_uom"
+                                />
```
```suggestion
                                <field
                                    name="product_uom_id"
                                    class="oe_inline"
                                    options="{'no_open': True, 'no_create': True}"
                                    groups="uom.group_uom"
                                />
                                <field name="product_uom_id" class="oe_inline" readonly="True" groups="!uom.group_uom" force_save="1"/>
```

victoralmau reviewed (changes requested):
Please remove migration scripts from 15.0 and fix test errors.

victoralmau commented on stock_request/tests/test_stock_request.py:
```diff
@@ -0,0 +1,1237 @@
+                ),
+            ],
+        }
+        cls.order = cls.request_order.new(vals)
```
```suggestion
        cls.order = cls.request_order.create(vals)
```

rousseldenis replied:
@victoralmau Sometimes, create is not necessary, especially in tests to improve performances.

rousseldenis commented:
@nicomacr Could you check tests ?

victoralmau commented:
Please, cherry-pick https://github.com/OCA/stock-logistics-warehouse/pull/1885 to the commit history

gcaceres93 commented:
Hello, is there any further advances on this pull request?

victoralmau commented:
Superseed by https://github.com/OCA/stock-logistics-request/pull/13

rousseldenis commented:
Closing as no answer for a long time
